src/20ec0_len_5040.c

```diff
@@ -176,68 +231,286 @@ INCLUDE_ASM(s32, "20ec0_len_5040", draw_all_status_icons);
 
 INCLUDE_ASM(s32, "20ec0_len_5040", create_status_icon_set);
 
-INCLUDE_ASM(s32, "20ec0_len_5040", remove_all_status_icons);
+void remove_all_status_icons(s32 iconID) {
+    HudStatusIcon* statusIcon = &D_800A0F44[iconID];
+
+    remove_status_debuff(iconID);
+    remove_status_static(iconID);
+    remove_status_transparent(iconID);
+    remove_status_4(iconID);
+    remove_status_icon_boost_jump(iconID);
+    remove_status_icon_boost_hammer(iconID);
+    remove_status_icon_boost_partner(iconID);
+    remove_status_icon_surprise(iconID);
+    remove_status_icon_peril(iconID);
+    remove_status_icon_danger(iconID);
+    statusIcon->flags = 0;
+}
 
 INCLUDE_ASM(s32, "20ec0_len_5040", set_status_icons_properties);
 
 INCLUDE_ASM(s32, "20ec0_len_5040", func_800477F4);
 
 INCLUDE_ASM(s32, "20ec0_len_5040", func_80047820);
 
-INCLUDE_ASM(s32, "20ec0_len_5040", remove_status_debuff);
+void remove_status_debuff(s32 iconID) {
+    HudStatusIcon* statusIcon = &D_800A0F44[iconID];
+
+    if (statusIcon->status1.active != FALSE && statusIcon->status1.removing == FALSE) {
```

Comment:
this can just be `if (statusIcon->status1.active && !statusIcon->status1.removing)` (bool checks don't need to expliticly compare with false)

Reply:
also relevant in below funcs